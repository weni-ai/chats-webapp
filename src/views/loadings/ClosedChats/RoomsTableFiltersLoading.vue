<template>
  <section
    class="rooms-table-filters-loading"
    :class="{ 'rooms-table-filters-loading--vertically': vertically }"
  >
    <div
      v-for="filter in 4"
      :key="`filter-${filter}`"
      class="rooms-table-filters-loading__filter"
    >
      <UnnnicSkeletonLoading />
      <UnnnicSkeletonLoading />
    </div>
    <UnnnicSkeletonLoading class="rooms-table-filters-loading__clear" />
  </section>
</template>

<script>
export default {
  name: 'RoomsTableFiltersLoading',

  props: {
    vertically: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.rooms-table-filters-loading {
  display: grid;
  grid-template-columns: repeat(4, 1fr) auto;

  gap: $unnnic-spacing-sm;
  justify-content: space-between;
  align-items: flex-end;

  overflow: hidden;

  &--vertically {
    grid-template-columns: auto;
    justify-content: stretch;

    .rooms-table-filters-loading__clear {
      width: 100%;
    }
  }

  &__filter {
    display: grid;
    grid-template-rows: 22px 46px;
    gap: $unnnic-spacing-nano;

    & > * {
      height: 100%;
    }
  }

  &__clear {
    height: 46px;
    width: 74px;
  }
}
</style>
