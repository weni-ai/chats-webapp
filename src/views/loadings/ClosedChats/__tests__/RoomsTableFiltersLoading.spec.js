import { mount } from '@vue/test-utils';
import RoomsTableFiltersLoading from '../RoomsTableFiltersLoading.vue';
import { beforeEach, describe, expect } from 'vitest';

const createWrapper = () => {
  return mount(RoomsTableFiltersLoading);
};

describe('RoomsTableFiltersLoading.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
