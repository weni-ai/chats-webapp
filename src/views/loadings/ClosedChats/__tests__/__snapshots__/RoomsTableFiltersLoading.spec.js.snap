// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoomsTableFiltersLoading.vue > to match snapshot 1`] = `
"<section data-v-7a440bf8="" class="rooms-table-filters-loading">
  <div data-v-7a440bf8="" class="rooms-table-filters-loading__filter"><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
  <div data-v-7a440bf8="" class="rooms-table-filters-loading__filter"><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
  <div data-v-7a440bf8="" class="rooms-table-filters-loading__filter"><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
  <div data-v-7a440bf8="" class="rooms-table-filters-loading__filter"><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-7a440bf8=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-7a440bf8="" class="rooms-table-filters-loading__clear"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
</section>"
`;
